src/extension_commands.ml

```diff
@@ -534,6 +534,143 @@ module Copy_type_under_cursor = struct
+        msg
+
+  let rec remove_duplicates = function
+    | (a : Custom_requests.Type_search.type_search_result) :: (b :: _ as t) ->
+      if String.equal a.name b.name && a.cost = b.cost then remove_duplicates t
+      else a :: remove_duplicates t
+    | unique -> unique
+
+  let get_search_results ~query ~limit ~with_doc ~position text_editor client =
+    let open Promise.Syntax in
+    let doc = TextEditor.document text_editor in
+    let uri = TextDocument.uri doc in
+    let* res =
+      Custom_requests.(
+        send_request
+          client
+          Type_search.request
+          (Type_search.make ~uri ~position ~limit ~query ~with_doc ()))
+    in
+    match res with
+    | [] ->
+      show_message `Info "Empty results";
+      [] |> Promise.return
+    | res -> remove_duplicates res |> Promise.return
+
+  let display_search_results results =
+    Vscode.(
+      Window.showQuickPickItems
+        ~choices:
+          (List.map
+             results
+             ~f:(fun (res : Custom_requests.Type_search.type_search_result) ->
+               ( Vscode.QuickPickItem.create
+                   ~label:res.name
+                   ~description:res.typ
+                   ~detail:(Option.value ~default:"" res.doc)
+                   ()
+               , res.name )))
+        ~options:
+          (QuickPickOptions.create
+             ~title:"Type/Polarity Search Results"
+             ~canPickMany:false
+             ~ignoreFocusOut:true
+             ())
+        ())
+
+  let _search_by_type =
+    let handler (instance : Extension_instance.t) ~args:_ =
+      let search_by_type () =
+        match Window.activeTextEditor () with
+        | None ->
+          Extension_consts.Command_errors.text_editor_must_be_active
+            extension_name
+            ~expl:"The command to search for a value by it's type"
+          |> show_message `Error "%s" |> Promise.return
+        | Some text_editor -> (
+          match Extension_instance.lsp_client instance with
+          | None ->
+            show_message `Warn "ocamllsp is not running" |> Promise.return
+          | Some (_client, ocaml_lsp)
+            when not (Ocaml_lsp.can_handle_search_by_type ocaml_lsp) ->
+            ocaml_lsp_doesnt_support_search_by_type instance ocaml_lsp
+            |> Promise.return
+          | Some (client, _) -> (
+            let open Promise.Syntax in
+            let position =
+              TextEditor.selection text_editor |> Selection.active
+            in
+            let* query_input =
+              Vscode.(
+                Window.showInputBox
+                  ~options:
+                    (InputBoxOptions.create
+                       ~title:"Search By Type"
+                       ~prompt:
+                         "Perform a search by type request by providing a type \
+                          signature to look for"
+                       ~placeHolder:"int -> string"
+                       ~value:"int -> string"
```

Comment:
I don't think we want a default value here. Additionally, maybe we should advertise the polarity-based syntax too.

The place holder could show both: `int -> string / -int +string`